### PR TITLE
CLI describe command

### PR DESCRIPTION
**Pull request by alex-zenml**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Added a `zenml [COMPONENT] describe` command (with the option to pass in the name of the component as an argument if you want something other than the current active component).

Works for `artifact`, `container_registry`, `metadata`, `orchestrator`, `stack`.

### CODE REVIEW

**Comment by alex-zenml** [comment]
@bcdurak 💣 is now defused. Ready for any reviews / feedback...

File: src/zenml/cli/metadata_store.py
Comment (schustmi) [inline src/zenml/cli/metadata_store.py:101]: I think it would be nicer to make this an `Optional[str]` and remove the default value in the click argument, and then set a default value if it's not passed similar to how you do it already: 
```python
metadata_store_name = metadata_store_name or repo.get_active_stack().metadata_store_name
```

**Review by htahir1 — COMMENTED** [review]
Nice work! I just have a question about the general approach: As all of these are `BaseComponent`, they have the `json()` and `dict()` methods to deserialize. Would it not be more robust to create one function that accepts a BaseComponent and then prints out a pretty YAML or JSON in a general way? That way this feature will be future proof to schema changes. WDYT

File: src/zenml/cli/metadata_store.py
Comment (schustmi) [inline src/zenml/cli/metadata_store.py:122]: The problem with this is that not every metadata store has a `uri` attribute (e.g. the MySQL one has host, port, ...).
I think we need to use a dynamic way to calculate what we print here using the actual attributes available for this specific metadata store (equivalent for all other stack components).

I think we can use pydantics `dict()` method to achieve at least a reasonable first implementation:
```python
for key, value in metadata_store.dict().items():
    cli_utils.declare(f"{key.upper()": {value}")
```

**Review by schustmi — CHANGES_REQUESTED** [review]
Looks good, just two small comments (I added them for metadata stores but should be applicable to all components)

File: src/zenml/cli/metadata_store.py
Comment (schustmi) [reply src/zenml/cli/metadata_store.py:122]: Also it prints "Active Metadata Store" even if I specify a name for a metadata store that is not active, maybe the title should just be the name of the store instead? And maybe a `cli_utils.declare()` to print whether its active or not?

Comment (htahir1) [reply src/zenml/cli/metadata_store.py:122]: I think your comments also align with my review. We need a way to generate this describe dynamically just like kubectl does it

**Comment by alex-zenml** [comment]
Thanks for the reviews @htahir1 and @schustmi. Indeed it's much better now. Fixed everything you mentioned.

**Review by AlexejPenner — APPROVED** [review]

**Review by schustmi — APPROVED** [review]

**Review by htahir1 — COMMENTED** [review]
It's good to merge now but I wonder if it can be even refactored to a generic function rather than the same for loop? If you refactor the loop into a function then we can also order the keys in alphabetical order and style etc without replacing it in three places. However we can merge if that's out of scope for the ticket

**Comment by alex-zenml** [comment]
@htahir1 I added in a function as suggested.

**Review by htahir1 — CHANGES_REQUESTED** [review]
Nice! Looks much cleaner! I noticed one small thing: We are also printing private properties here like `_SUPERFLUOUS_OPTIONS`. I would add the following to avoid it:

```
from zenml.core.base_component import SUPERFLUOUS_OPTIONS_ATTRIBUTE_NAME

cli_utils.print_component_properties(COMPONENT.dict(exclude={SUPERFLUOUS_OPTIONS_ATTRIBUTE_NAME}))
```

**Comment by schustmi** [comment]
The issue with private properties is fixed on my kubeflow branch already, where I filter the dict for all base components (so I think no need to fix it here, should be fine once both are merged)

**Comment by htahir1** [comment]
Ok then Lets Go To Main @strickvl !

**Pull request merged**
